builder/builder-source-archive.c

```diff
@@ -426,6 +427,24 @@ unrpm (GFile   *dir,
   return res;
 }
 
+static gboolean
+undeb (GFile   *dir,
+       const char *deb_path,
+       GError **error)
+{
+  gboolean res;
+  const gchar *argv[] = { "sh", "-c", NULL, NULL };
+  char *undeb_cmdline = g_strdup_printf("ar t %s | grep data.tar | xargs ar x %s",
```

Comment:
This contains two instances of shell injection. I'm about to send a PR to fix the equivalent bug in the RPM code path.

On systems that have `dpkg`, it's much better to unpack debs with `dpkg-deb --fsys-tarfile` (analogous to rpm2cpio) or `dpkg-deb --extract`, rather than hard-coding the knowledge that a `.deb` is an ar archive containing some sort of tarball.